README.md

```diff
@@ -344,6 +344,31 @@ These settings can be changed at any time.
 * `settings.updates.version-lock`: Controls the version that will be selected when you issue an update request.  Can be locked to a specific version like `v1.0.0`, or `latest` to take the latest available version.  Defaults to `latest`.
 * `settings.updates.ignore-waves`: Updates are rolled out in waves to reduce the impact of issues.  For testing purposes, you can set this to `true` to ignore those waves and update immediately.
 
+#### Network settings
+
+These settings will configure the networking behavior of the following services:
+* For all variants:
+    * [containerd.service](packages/containerd/containerd.service)
+    * [host-containerd.service](packages/host-ctr/host-containerd.service)
+* For Kubernetes variants:
+    * [kubelet.service](packages/kubernetes-1.18/kubelet.service)
+* For ECS variants:
+    * [docker.service](packages/docker-engine/docker.service)
+    * [ecs.service](packages/ecs-agent/ecs.service)
+
+* `settings.network.http-proxy`: The HTTP proxy server to be used by services listed above.
+* `settings.network.https-proxy`: The HTTPS proxy server to be used by services listed above.
+* `settings.network.no-proxy`: A list of hosts that are excluded from proxying.
+
+`NO_PROXY` and `no_proxy` will always include `localhost`, `127.0.0.1`, and `169.254.169.254`
+(Instance Metadata Service endpoint)
+
+If you are running Kubernetes variants, `NO_PROXY` and `no_proxy` will be appended with the Kubernetes API server endpoint
+and other commonly used Kubernetes DNS suffixes to facilitate intra-cluster networking.
+
+If you are running the Amazon ECS variants, `NO_PROXY` will be appended with the IMDS endpoint, the Task Metadata Endpoint
+ and the docker socket path.
```

Comment:
```suggestion
If you're running the Amazon ECS variant, the no-proxy list will automatically include the task metadata endpoint and the Docker socket path.
```

Reply:
We already mentioned instance metadata.  It raises another question, though - should we include localhost in the ECS list?  We (rightfully) say it'll be on the list, but it's not in ecs.config.

Maybe a better question - does ecs-agent respect no-proxy environment variables?  Can we have ecs.service source the environment file instead of setting up proxying in ecs.config?  (It'd set _additional_ variables, perhaps, but that shouldn't matter.)

If we can't do that, maybe we should add localhost to ecs.config, for consistency.

Reply:
> Maybe a better question - does ecs-agent respect no-proxy environment variables?

Yes

> Can we have ecs.service source the environment file instead of setting up proxying in ecs.config?

Yes, the `ecs.config` file is just sourced as an environment file in the systemd unit anyway.

Reply:
Is there any harm in adding the task metadata endpoint to the no-proxy list even when it's a Kubernetes variant or any variant other than ECS?

Letting `ecs.config` set the proxy environment variables has the nice property of separating out the ECS specific no-proxy entries to just the ECS agent service.

Reply:
There could possibly be harm if someone is using that link-local address for their own purposes, but you'd have to go out of your way.  I think it'd be reasonable to have everything in one list.